OpenDreamRuntime/DreamThread.cs

```diff
@@ -203,7 +212,41 @@ public sealed class DreamThread {
                 CurrentlyExecuting.Value!.Push(this);
                 while (_current != null) {
                     // _current.Resume may mutate our state!!!
-                    switch (_current.Resume()) {
+
+                    bool TryCatchException(DreamValue value) {
+                        if (!_stack.Any(x => x.IsCatching())) return false;
+
+                        while (!_current.IsCatching()) {
+                            PopProcState();
+                        }
+                        _current.CatchException(value);
+                        return true;
+                    }
+
+                    ProcStatus status;
+                    try {
+                        status = _current.Resume();
+                    } catch (DMCrashRuntime dmCrashRuntime) {
+                        //skip one level on the call stack because crash is being treated as an actual proc
+                        PopProcState();
+                        if (TryCatchException(dmCrashRuntime.Value)) continue;
+                        HandleException(dmCrashRuntime);
+                        status = ProcStatus.Returned;
+                    } catch (DMRuntime dmRuntime) {
+                        if (TryCatchException(dmRuntime.Value)) continue;
+                        HandleException(dmRuntime);
+                        status = ProcStatus.Returned;
+                    } catch (DMError dmError) {
+                        CancelAll();
+                        HandleException(dmError);
+                        status = ProcStatus.Cancelled;
+                    } catch (Exception exception) {
+                        if (TryCatchException(new DreamValue(exception.Message))) continue;
```

Comment:
BYOND appears to turn any exception that isn't from a `throw` into an [/exception object](https://www.byond.com/docs/ref/#/exception).